rpcs3/Emu/RSX/VK/VKFragmentProgram.cpp

```diff
@@ -207,6 +213,19 @@ void VKFragmentDecompilerThread::insertMainStart(std::stringstream & OS)
 				vk::insert_fog_declaration(OS, m_prog.fog_equation);
 				return;
 			}
+			
+			if (m_prog.front_back_color_enabled)
+			{
+				if (PI.name == "spec_color" && m_prog.back_color_specular_output)
+				{
+					OS << "	vec4 spec_color = gl_FrontFacing ? vec4(front_spec_color) : vec4(spec_color);\n";
+				}
+				if (PI.name == "diff_color" && m_prog.back_color_diffuse_output)
+				{
+					OS << "	vec4 diff_color = gl_FrontFacing ? vec4(front_diff_color) : vec4(diff_color);\n";
+				}
+				return;
```

Comment:
This is the  spriv generated and the color looking in this build .


[FragmentProgram.spirv](https://github.com/RPCS3/rpcs3/files/344549/FragmentProgram.txt)